NERODevelopment/src/controllers/offViewController.cpp

```diff
@@ -0,0 +1,54 @@
+#include "offviewcontroller.h"
+
+OffViewController::OffViewController(OffViewModel *viewModel, QObject *parent)
+    : QObject(parent), m_viewModel(viewModel)
+{
+    updateAttributesFromModel();
+}
+
+QMap<QString, AttributeStatus> OffViewController::getAttributeStatus() const
+{
+    return m_attributeStatus;
+}
+
+
+
+QMap<QString, AttributeStatus> OffViewController::getAttributeStatus() const { return m_attributeStatus; }
+void OffViewController::setAttributeStatus(const QString &name, AttributeStatus status)
+{
+    if (m_attributeStatus.value(name) != status)
+    {
+        m_attributeStatus[name] = status;
+        emit attributeStatusChanged(m_attributeStatus);
+    }
+}
+
+void OffViewController::updateAttributesFromModel() {
+    for (const auto& attributeName : {SIDEBRBS, BMS, BSPD, MPU, BOTS, INERTIA, CPBRB, TSMS, HVDINTRLK, HVCNCTR}) {
+        setAttributeStatus(attributeName, m_attributeStatus[attributeName]);
```

Comment:
this is just going to set it to itself isnt it? You need to get the updated value from the model. You can use the getById Function I believe. Actually you can add the attribute names macros to the data_type_names.h macro file